include/dlaf/types.h

```diff
@@ -36,6 +37,16 @@ enum class Device {
 #endif
 };
 
+inline std::ostream& operator<<(std::ostream& os, const Device& device) {
+  if (device == Device::CPU) {
+    os << "CPU";
+  }
+  else if (device == Device::GPU) {
+    os << "GPU";
+  }
+  return os;
+}
```

Comment:
Yeah, true! For some of these I found `if-else` easier because the logic was a bit more complicated, but in this case it definitely makes sense to use `switch`. I'll check if there are others as well.

Reply:
I've changed the two `operator<<` overloads in this file to use `switch`. I've left a number of `if-else` versions in `options.h` that I find more readable that way. Let me know if you think they'd benefit from using `switch` as well.